libsql-server/src/connection/libsql.rs

```diff
@@ -76,60 +60,62 @@ where
     ) -> Result<Self> {
         let txn_timeout = config_store.get().txn_timeout.unwrap_or(TXN_TIMEOUT);
 
-        let mut this = Self {
-            db_path,
-            stats,
-            broadcaster,
-            config_store,
-            extensions,
-            max_response_size,
-            max_total_response_size,
-            auto_checkpoint,
-            current_frame_no_receiver,
-            _db: None,
-            wal_wrapper,
-            encryption_config,
-            block_writes,
-            resolve_attach_path,
-            connection_manager: ConnectionManager::new(txn_timeout),
-            make_wal_manager,
-        };
-
-        let db = this.try_create_db().await?;
-        this._db = Some(db);
-
-        Ok(this)
-    }
+        let (snd, recv) = async_channel::bounded(16);
+        tokio::spawn(async move {
+            let inner = MakeLibSqlConnInner {
+                db_path,
+                stats,
+                broadcaster,
+                config_store,
+                extensions,
+                max_response_size,
+                max_total_response_size,
+                auto_checkpoint,
+                current_frame_no_receiver,
+                wal_wrapper,
+                encryption_config,
+                block_writes,
+                resolve_attach_path,
+                connection_manager: ConnectionManager::new(txn_timeout),
+                make_wal_manager,
+            };
 
-    /// Tries to create a database, retrying if the database is busy.
-    async fn try_create_db(&self) -> Result<LibSqlConnection<W>> {
-        // try 100 times to acquire initial db connection.
-        let mut retries = 0;
-        loop {
-            match self.make_connection().await {
-                Ok(conn) => return Ok(conn),
-                Err(
-                    err @ Error::RusqliteError(rusqlite::Error::SqliteFailure(
-                        rusqlite::ffi::Error {
-                            code: ErrorCode::DatabaseBusy,
-                            ..
-                        },
-                        _,
-                    )),
-                ) => {
-                    if retries < 100 {
-                        tracing::warn!("Database file is busy, retrying...");
-                        retries += 1;
-                        tokio::time::sleep(Duration::from_millis(100)).await
-                    } else {
-                        Err(err)?;
-                    }
+            loop {
+                let conn = inner.make_connection().await;
```

Comment:
Just asking for the sake of clarity: it's not a connection pool in sense that we don't return anything to that pool (should we?) - we just want to bootstrap few connections in a background in order to remove initialization delay from request latencies.

BTW - after first 16 connections we will create them one by one. Should we try to maintain "free pool" size at certain level in order to guarantee same behaviour for startup and after it for sqld?

Reply:
Yeah not really a pool in the sense that connections are reused.

Definitely, the pool size will be configurable, and I'm also thinking about making it grow for namespaces that see load, something like "if the queue ever goes empty, bump it's size"

Reply:
Also, as soon as a connection is popped, a new one is added. The loop tries to maintain a full queue